synapse/http/client.py

```diff
@@ -782,12 +783,15 @@ def connectionLost(self, reason: Failure) -> None:
             self.deferred.errback(reason)
 
 
-def readBodyToFile(
+def read_body_with_max_size(
```

Comment:
I wanted to make this async and handle the `Deferred` logic internally, but the call from the `MatrixFederationHttpClient` adds a timeout, which involves having both a timeout and a reactor. The timeout could be made optional easily, but needed both or neither is kind of meh.